### PR TITLE
Track metrics on appointment notifications

### PR DESCRIPTION
**Pull request by rsanheim**
**Story card:** [ch3300](https://app.clubhouse.io/simpledotorg/story/3300/as-a-developer-i-want-to-have-a-working-and-verifiable-sbx-twilio-setup)

## Because

We want to be able to track Appt notification success / error rates over time via Datadog.

## This addresses

tracking those metrics

### CODE REVIEW

File: app/jobs/appointment_notification/worker.rb [hunk 1]
```diff
@@ -6,14 +6,33 @@ class AppointmentNotification::Worker
 
   DEFAULT_LOCALE = :en
 
+  class Metrics
```
Comment (rsanheim) [inline app/jobs/appointment_notification/worker.rb:9]: Going to leave this as an inner class until I have a use for this sort of thing elsewhere, which will probably be soon.

Comment (rsanheim) [reply app/jobs/appointment_notification/worker.rb]: never mind, extracted already.

File: app/jobs/appointment_notification/worker.rb [hunk 2]
```diff
@@ -6,14 +6,22 @@ class AppointmentNotification::Worker
...
   DEFAULT_LOCALE = :en
 
+  def metrics
+    @metrics ||= Metrics.with_object(self)
+  end
+
   def perform(appointment_id, communication_type, locale = nil)
     appointment = Appointment.find_by(id: appointment_id)
     unless appointment
+      metrics.increment(:skipped)
```
Comment (harimohanraj89) [inline app/jobs/appointment_notification/worker.rb:16]: Consider `skipped.missing_appointment`, and `skipped.previously_communicated` below?

File: spec/controllers/api/v3/exotel_call_sessions_controller_spec.rb [hunk 3]
```diff
@@ -123,7 +123,7 @@
         expect(response).to have_http_status(200)
       end
 
-      it "should report metrics to new relic" do
```
Comment (harimohanraj89) [inline spec/controllers/api/v3/exotel_call_sessions_controller_spec.rb:126]: Lol, nice.

**Review by harimohanraj89 — APPROVED** [review]
Some `standard` stuff and nonblocking suggestions. Looks good to me.

**Pull request merged**